File: app/src/test/java/com/mgaetan89/showsrage/fragment/ShowFragment_GetSeasonsSortTest.java
package com.mgaetan89.showsrage.fragment;

import android.content.SharedPreferences;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.Parameterized;

import java.util.Arrays;
import java.util.Collection;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

@RunWith(Parameterized.class)
public class ShowFragment_GetSeasonsSortTest {
	@Parameterized.Parameter(1)
	public String sort;

	@Parameterized.Parameter(0)
	public boolean status;

	private SharedPreferences preferences = null;

	@Before
	public void before() {
		this.preferences = mock(SharedPreferences.class);
		when(this.preferences.getBoolean("display_seasons_sort", false)).thenReturn(this.status);
	}

	@Test
	public void getSeasonsSort() {
		String sort = ShowFragment.getSeasonsSort(this.preferences);

		assertThat(sort).isEqualTo(this.sort);
	}

	@Test
	public void getSeasonsSortNull() {
		String sort = ShowFragment.getSeasonsSort(null);

		assertThat(sort).isEqualTo("desc");
	}

	@After
	public void after() {
		this.preferences = null;
	}

	@Parameterized.Parameters
	public static Collection<Object[]> data() {
		return Arrays.asList(new Object[][]{
				{false, "desc"},
				{true, "asc"}
		});
	}
}

File: app/src/test/kotlin/com/mgaetan89/showsrage/fragment/ShowFragment_GetSeasonsSortTest.kt
package com.mgaetan89.showsrage.fragment

import android.content.SharedPreferences
import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.Parameterized
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

@RunWith(Parameterized::class)
class ShowFragment_GetSeasonsSortTest(val status: Boolean, val sort: String) {
    private lateinit var preferences: SharedPreferences

    @Before
    fun before() {
        this.preferences = mock(SharedPreferences::class.java)
        `when`(this.preferences.getBoolean("display_seasons_sort", false)).thenReturn(this.status)
    }

    @Test
    fun getSeasonsSort() {
        val sort = ShowFragment.getSeasonsSort(this.preferences)

        assertThat(sort).isEqualTo(this.sort)
    }

    @Test
    fun getSeasonsSortNull() {
        val sort = ShowFragment.getSeasonsSort(null)

        assertThat(sort).isEqualTo("desc")
    }

    companion object {
        @JvmStatic
        @Parameterized.Parameters
        fun data(): Collection<Array<Any>> {
            return listOf(
                    arrayOf(false, "desc"),
                    arrayOf(true, "asc")
            )
        }
    }
}
